src/compiler/utils/mapped_code.ts

```diff
@@ -251,6 +251,11 @@ export function combine_sourcemaps(
 
 	if (!map.file) delete map.file; // skip optional field `file`
 
+	// When source maps are combined and the leading map is empty, sources is not set.
+	// Add the filename to the empty array in this case. 
+	// Possibly a bug in the remapping library https://github.com/ampproject/remapping/issues/116
```

Comment:
I'm not sure I'd say "Possibly a bug" as it's not quite clear to me that there's any bug there and the bad behavior seems to be more on the Svelte side, but maybe "Possibly further improvements to the remapping library may help address this as well"

Reply:
Yeah I wrote that when things were not as clear, that comment should be updated

Reply:
```suggestion
	// Further improvements to remapping may help address this as well https://github.com/ampproject/remapping/issues/116
```